internal/moreos/proc_linux_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package moreos
+
+import (
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestCmdStart tests CmdStart by starting a parent process that spawn a child process and then sends
+// signal to terminate the child process which in effect will terminate the parent process via
+// the configured Pdeathsig.
+func TestCmdStart(t *testing.T) {
+	tempDir := t.TempDir()
+
+	cmd := newCmdForTest("TestCmdStart_Parent", tempDir)
+	err := cmd.Run()
+	require.NoError(t, err)
+
+	require.FileExists(t, filepath.Join(tempDir, "parent-sigterm"))
+	require.FileExists(t, filepath.Join(tempDir, "child-sigterm"))
+
+	p, err := os.Stat(filepath.Join(tempDir, "parent-sigterm"))
+	require.NoError(t, err)
+	c, err := os.Stat(filepath.Join(tempDir, "child-sigterm"))
+	require.NoError(t, err)
+
+	// Make sure child process receives SIGTERM right after the parent process.
+	require.True(t, c.ModTime().After(p.ModTime()) || c.ModTime().Equal(p.ModTime()))
+}
+
+// TestCmdStart_Parent starts a child process (TestChildProcess), wait for 1 second and send SIGTERM
+// to the parent process.
+func TestCmdStart_Parent(t *testing.T) {
+	if os.Getenv("FUNC_E_TEST_PROCESS") != "1" {
+		return
+	}
+
+	tempDir := os.Getenv("FUNC_E_TEST_DIR")
+	cmd := newCmdForTest("TestCmdStart_Child", tempDir)
+	// Set Pdeathsig for this command.
+	cmd.SysProcAttr = ProcessGroupAttr()
+
+	// Parent process hints.
+	pid := os.Getpid()
+	tid := syscall.Gettid()
```

Comment:
I think in the doc you say the use case is pid or did, but really this isn't generic, we only use it for pid, and it is internal anyway. Let's correct the docs to say only pid, and generally not document uses or design for uses we don't use or only use in tests. That is more honest and the more specific we are the easier it is to review for simplifications.

Reply:
I'll use `syscall.Kill(pid, syscall.SIGTERM)` at the bottom when sending the signal to the parent process, does it make sense?